### PR TITLE
Add `mapper` argument

### LINKED ISSUE #4: Consider accepting an optional `mapper` argument
Would something like this be acceptable?
```js
const pProps = require('p-props');
const got = require('got');

const fetch = url => got(url).then(res => res.body);

const sites = {
	ava: 'ava.li',
	todomvc: 'todomvc.com',
	github: 'github.com'
};

pProps(sites, fetch).then(result => {
	console.log(result);
	//=> {ava: '<!doctype ...', todomvc: '<!doctype ...', github: '<!doctype ...'}
});
```

### PR DESCRIPTION
**Pull request by wtgtybhertgeghgtwtg**
Closes #4.

```js
const pProps = require('p-props');
const got = require('got');

const mapper = url => got.head(url).then(res => res.requestUrl);

const sites = {
	sindresorhus: getWebsiteFromUsername('sindresorhus'), //=> Promise
	ava: 'ava.li'
};

pProps(sites, mapper).then(result => {
	console.log(result);
	//=> {sindresorhus: 'http://sindresorhus.com/', ava: 'http://ava.li/'}
});
```

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -36,16 +36,34 @@ pProps(sites).then(result => {
...
 
+#### mapper(element, index)
+
+Type: `Function`
+
+Expected to return a `Promise` or value.
+
+#### options
+
+Type: `Object`
```
Comment (kevva) [inline readme.md]: Could just link to the [`p-map` options](https://github.com/sindresorhus/p-map#options) so we don't have to keep updating them in here.

File: package.json [hunk 2]
```diff
@@ -39,5 +39,8 @@
     "ava": "*",
     "delay": "^2.0.0",
     "xo": "*"
+  },
+  "dependencies": {
```
Comment (kevva) [inline package.json]: This should be listed before `devDependencies`.

**Comment by wtgtybhertgeghgtwtg** [comment]
So, looking at this again, should `mapper` take `key` as a second argument?

**Comment by kevva** [comment]
> So, looking at this again, should mapper take key as a second argument?

Hmm, maybe it could take `([key, value], index)` as arguments? Like a `Map`.

**Comment by wtgtybhertgeghgtwtg** [comment]
`index` doesn't really have much of a meaning here, though.  It'd also require the use to destructure the first argument.  So, a (presumably) common use case like
```
pProps(object, value => asyncFunction(value))
```
would become
```
pProps(object, ([,value]) => asyncFunction(value))
```

**Comment by wtgtybhertgeghgtwtg** [comment]
I can't think of a good way to do this without `Object.entries`.  Should I hold off until that's available?

**Comment by wtgtybhertgeghgtwtg** [comment]
This didn't occur to me until just now, but is there a way I should indicate in the documentation that `mapper` is optional?

**Comment by sindresorhus** [comment]
> This didn't occur to me until just now, but is there a way I should indicate in the documentation that mapper is optional?

I use this syntax `foo(bar, [mapper])` to indicate that something is optional. It's a fairly common pattern.

**Comment by sindresorhus** [comment]
This looks great. Thank you @wtgtybhertgeghgtwtg 🙌

**Pull request merged**